exercises/list-ops/src/ListOps.hs

```diff
@@ -13,25 +13,25 @@ import Prelude hiding
   ( length, reverse, map, filter, foldr, (++), concat )
 
 foldl' :: (b -> a -> b) -> b -> [a] -> b
-foldl' = error "You need to implement this function."
+foldl' f x xs = error "You need to implement this function."
 
 foldr :: (a -> b -> b) -> b -> [a] -> b
-foldr = error "You need to implement this function."
+foldr f x xs = error "You need to implement this function."
 
 length :: [a] -> Int
-length = error "You need to implement this function."
+length xs = error "You need to implement this function."
 
 reverse :: [a] -> [a]
-reverse = error "You need to implement this function."
+reverse xs = error "You need to implement this function."
 
 map :: (a -> b) -> [a] -> [b]
-map = error "You need to implement this function."
+map f xs = error "You need to implement this function."
 
 filter :: (a -> Bool) -> [a] -> [a]
-filter = error "You need to implement this function."
+filter f xs = error "You need to implement this function."
```

Comment:
I'm used to `p` for _predicate_ as well, but thought `f` would make it more clear that this is a function. Happy to change 👍